compiler/damlc/tests/src/DA/Test/DamlcUpgrades.hs

```diff
@@ -555,13 +642,19 @@ tests damlc =
             writeFiles dir (projectFile "mylib-v2" (Just depDar) : newVersion)
             writeFiles depDir (projectFile "mylib-v1" Nothing : oldVersion)
             callProcessSilent damlc ["build", "--project-root", depDir, "-o", depDar]
-            case expectedError of
-              Nothing ->
+            case expectation of
+              Succeed ->
                   callProcessSilent damlc ["build", "--project-root", dir, "-o", dar]
-              Just regex -> do
+              FailWithError regex -> do
                   stderr <- callProcessForStderr damlc ["build", "--project-root", dir, "-o", dar]
-                  unless (matchTest (makeRegex regex :: Regex) stderr) $
-                      assertFailure ("Regex '" <> show regex <> "' did not match stderr:\n" <> show stderr)
+                  let regexWithSeverity = "Severity: DsError\nMessage: \n" <> regex
```

Comment:
These changes are the other side of adding warnings: we now check for success, failure with an error message, or success with a warning message.